spec-tests/src/test/java/org/candlepin/spec/jobs/JobScheduleSpecTest.java

```diff
@@ -80,19 +81,14 @@ public void shouldPurgeExpiredPools() throws Exception {
         OwnerApi ownerApi = client.owners();
         PoolsApi poolsApi = client.pools();
         OwnerDTO owner = ownerApi.createOwner(Owners.random());
-        ProductDTO product = new ProductDTO();
-        product.setId(StringUtil.random("ID"));
-        product.setName(StringUtil.random("Test Product"));
+        ProductDTO product = Products.random();
         product = client.ownerProducts().createProductByOwner(owner.getKey(), product);
 
-        PoolDTO pool = new PoolDTO();
-        pool.setProductId(product.getId());
-        pool.setSubscriptionId(StringUtil.random("source_sub"));
```

Comment:
Not setting the subscriptionId field for both these tests has fixed the issue in this case, but we should preemptively make a change to not allow this kind of flakiness to happen in the future: we can delete the `StringUtil.random` method and rename `StringUtil.randomSuffix` into `StringUtil.random`, and use that throughout the code, since it will produce a prefix with shorter/limited length

Reply:
I will make a separate branch for this.